### ISSUE TITLE
Zero-shot classification prompt template

### ISSUE THREAD

**Issue by StijnvanderLippe** [Imageomics/bioclip-2#32]
Hi all,

Thanks for the great model! I'm using BioCLIP 2.5 to classify species from camera traps. I had a question regarding how to supply the class labels to the model. My current code is given below, where my class labels in `SPECIES_LIST` is a list of scientific species names (e.g., `Dama dama`, or `Capreolus capreolus`), which are encoded using the tokenizer.

When classifying, the image is encoded as well and we compute the most likely species for the image. My question is if I instead of using just the scientific name, what would your recommendation be on how to supply the class labels? Should I use a prompt templates such as `Image of a <class>` and use those as (encoded) class labels for improved results?

I've tried searching through the repo to identify how the model was trained but could not figure it out with certainty. Hence my question here.
Thanks in advance!

```python
class AnimalClassifier:
    """
    Animal species classifier using CLIP-like zero-shot classification.

    Attributes
    ----------
    text_features : list[str]
        Encoded scientific names used as zero-shot labels.
    """

    def __init__(self, confidence_threshold: float) -> None:
        """Initialize CLIP-like classifier."""
        self.model, _, self.preprocess = open_clip.create_model_and_transforms("hf-hub:imageomics/bioclip-2.5-vith14")
        self.tokenizer = open_clip.get_tokenizer(f"hf-hub:imageomics/bioclip-2.5-vith14")
        self.model.eval()

        self.text_features = self._encode_species(SPECIES_LIST)

    def _encode_species(self, species_list: list[str]) -> torch.Tensor:
        """
        Pre-encode species labels for efficient classification.

        Parameters
        ----------
        species_list : list[str]
            Scientific names used as zero-shot labels.

        Returns
        -------
        torch.Tensor
            Encoded scientific names.
        """
        # Should this instead be a prompt template like 'Image of a <class>'?
        text = self.tokenizer(species_list)
        with torch.no_grad():
            text_features = self.model.encode_text(text)
            text_features = text_features / text_features.norm(dim=-1, keepdim=True)
        return text_features

    def classify(self, image: Path | Image.Image) -> tuple[str, float]:
        """
        Classify image into configured species labels.

        Parameters
        ----------
        image : Path | Image.Image
            Path to image file, or PIL image in RGB format.

        Returns
        -------
        species : str
            Most likely species.
        conf : float
            Classification confidence.
        """
        if isinstance(image, Path):
            with Image.open(image) as image_file:
                image = image_file.convert("RGB")

        try:
            image_tensor = self.preprocess(image).unsqueeze(0)

            with torch.no_grad():
                image_features = self.model.encode_image(image_tensor)
                image_features = image_features / image_features.norm(dim=-1, keepdim=True)

                logits = (image_features @ self.text_features.T).squeeze(0)
                probs = torch.nn.functional.softmax(logits * 100, dim=-1)
                confidence, idx = torch.max(probs, dim=0)

            species = SPECIES_LIST[idx.item()]
            conf = confidence.item()
            return species, conf
```

**Comment by vimar-gu** [comment]
Hello there!

It should be using templates. We have a public Python package `pybioclip` and you can refer to https://imageomics.github.io/pybioclip/python-tutorial/#predict-from-a-list-of-classes for predicting from custom species lists.

**Comment by StijnvanderLippe** [comment]
Thanks for the answer! I will take a look at the linked code.
